app/models/call_data_record.rb

```diff
@@ -2,8 +2,10 @@ class CallDataRecord < ApplicationRecord
   extend Enumerize
 
   enumerize :direction, in: %i[inbound outbound]
+  enumerize :call_leg, in: %i[A B]
+
   belongs_to :phone_call
   has_one_attached :file
 
-  validates :file, presence: true
+  validates :file, :call_leg, presence: true
```

Comment:
We don't need to validate presence here. Schema is not null and we also have enumerize

Reply:
We may don't need it, but it's better to have it when we can know the invalid fields in the errors collection when the object is used.

Reply:
Not sure because we have alot of fields that are not null but we don't have validates presence.